stubs/ldap3/ldap3/core/connection.pyi

```diff
@@ -0,0 +1,169 @@
+    open: Any
+    get_response: Any
+    post_send_single_response: Any
+    post_send_search: Any
+    def __init__(
+        self,
+        server: Server | str,
+        user: str | None = ...,
+        password: str | None = ...,
+        auto_bind: Literal["DEFAULT", "NONE", "NO_TLS", "TLS_BEFORE_BIND", "TLS_AFTER_BIND"] = ...,
+        version: int = ...,
+        authentication: Literal["ANONYMOUS", "SIMPLE", "SASL", "NTLM"] | None = ...,
+        client_strategy: Literal[
+            "SYNC", "SAFE_SYNC", "ASYNC", "LDIF", "RESTARTABLE", "REUSABLE", "MOCK_SYNC", "MOCK_ASYNC", "ASYNC_STREAM"
+        ] = ...,
+        auto_referrals: bool = ...,
+        auto_range: bool = ...,
+        sasl_mechanism: str | None = ...,
+        sasl_credentials: Any | None = ...,
+        check_names: bool = ...,
+        collect_usage: bool = ...,
+        read_only: bool = ...,
+        lazy: bool = ...,
+        raise_exceptions: bool = ...,
+        pool_name: str | None = ...,
+        pool_size: str | None = ...,
+        pool_lifetime: int | None = ...,
+        cred_store: Any | None = ...,
+        fast_decoder: bool = ...,
+        receive_timeout: Any | None = ...,
+        return_empty_attributes: bool = ...,
+        use_referral_cache: bool = ...,
+        auto_escape: bool = ...,
+        auto_encode: bool = ...,
+        pool_keepalive: Any | None = ...,
+        source_address: str | None = ...,
+        source_port: int | None = ...,
+        source_port_list: Any | None = ...,
+    ) -> None: ...
+    def repr_with_sensitive_data_stripped(self): ...
+    @property
+    def stream(self): ...
+    @stream.setter
+    def stream(self, value) -> None: ...
+    @property
+    def usage(self): ...
+    def __enter__(self) -> Connection: ...
+    def __exit__(
+        self, exc_type: Type[BaseException] | None, exc_val: BaseException | None, exc_tb: TracebackType | None
+    ) -> Literal[False] | None: ...
+    def bind(self, read_server_info: bool = ..., controls: Any | None = ...): ...
+    def rebind(
+        self,
+        user: Any | None = ...,
+        password: Any | None = ...,
+        authentication: Any | None = ...,
+        sasl_mechanism: Any | None = ...,
+        sasl_credentials: Any | None = ...,
+        read_server_info: bool = ...,
+        controls: Any | None = ...,
+    ): ...
+    def unbind(self, controls: Any | None = ...): ...
+    def search(
+        self,
+        search_base: str,
+        search_filter: str,
+        search_scope: Literal["BASE", "LEVEL", "SUBTREE"] = ...,
+        dereference_aliases: Literal["NEVER", "SEARCH", "FINDING_BASE", "ALWAYS"] = ...,
+        attributes: Any | None = ...,
+        size_limit: int = ...,
+        time_limit: int = ...,
+        types_only: bool = ...,
+        get_operational_attributes: bool = ...,
+        controls: Any | None = ...,
+        paged_size: int | None = ...,
+        paged_criticality: bool = ...,
+        paged_cookie: str | None = ...,
```

Comment:
Perhaps this can be bytes too? It is passed to a function named `escape_bytes()`.

Reply:
+1 It's also passed to `OctetString` from `pyasn1`, which also accepts `bytes`.